sdk/include/sdk/Node.h

```diff
@@ -17,16 +17,28 @@
 #ifndef VEHICLE_APP_SDK_NODE_H
 #define VEHICLE_APP_SDK_NODE_H
 
+#include <memory>
 #include <string>
+#include <unordered_map>
 
 namespace velocitas {
 
+class DataPoint;
+
 /**
  * @brief A tree node.
  *
  */
 class Node {
 public:
+    enum class Type {
+        BRANCH,
+        UNKNOWN_LEAF_TYPE,
```

Comment:
There exists in COVESA VSS also nodes for structs and properties. Any particular reason to limit this one to leaf types, or could we have it more general as `UNKNOWN_TYPE`?

Reply:
Having spent some time to think about this: I don't like defining an `unknown` for enums so much, as it always raises the question how to handle this value.
I introduced the `UNKNOWN_LEAF_TYPE` just for backwards compatibility reasons, means in case the SDK is used by a model generated before doing these extensions.

I've a question regarding your hint regarding structs and properties. At least for structs my understanding is that this isn't a node type but a data type: A struct represents a datapoint/signal of complex nature and can be either of type attribute, sensor, or actuator. I.e. those node types shall not be mixed within a single struct type.

Reply:
Yes, the struct/property VSS nodes will never appear in the same tree as signals and branches. 

Thanks for the explanation. Possibly add a comment on `UNKNOWN_LEAF_TYPE` that it is there just for backward compatibility reasons

Reply:
Commented